### PR TITLE
[IE TESTS][Conformance] API conformance instances

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - API tests were instantiated for API Conformance suite without plugin specific config parameters

### Tickets:
 - *58722*

### CODE REVIEW

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,9 @@ addIeTargetTest(
         ADDITIONAL_SOURCE_DIRS
             ${CMAKE_CURRENT_SOURCE_DIR}/src
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/CMakeLists.txt:11]: do we need the renaming above `avior/executable_network/get_metric.cpp → ...r/executable_network/core_integration.cpp `?

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        resultConfig.push_back(configItem);
+    }
+    return resultConfig;
+}
+
+inline const std::string generateComplexDeviceName(const std::string& deviceName) {
+    return deviceName + ":" + ConformanceTests::targetDevice;
+}
+
+inline const std::vector<std::string> returnAllPosibleDeviceCombination() {
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp:71]: Posible -> Possible

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            { "GPU", "clDNNPlugin" },
+            { "HETERO", "HeteroPlugin" },
+            { "MULTI", "MultiDevicePlugin" },
+            { "MYRIAD", "myriadPlugin" },
+            { "TEMPLATE", "templatePlugin" },
+    };
+    if (devices.find(deviceName) == devices.end()) {
+        throw std::runtime_error("Incorrect device name");
+    }
+    return devices.at(deviceName);
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp:34]: is it used? for what?

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/caching/caching_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                            ::testing::Combine(
+                                    ::testing::ValuesIn(LoadNetworkCacheTestBase::getStandardFunctions()),
+                                    ::testing::ValuesIn(cachingElemTypes),
+                                    ::testing::ValuesIn(cachingBatchSizes),
+                                    ::testing::Values(ConformanceTests::targetDevice)),
+                            LoadNetworkCacheTestBase::getTestCaseName);
+
+    INSTANTIATE_TEST_SUITE_P(smoke_CachingSupportCase_CPU_Internal, LoadNetworkCacheTestBase,
+                            ::testing::Combine(
+                                    ::testing::ValuesIn(internal_functions_cpu()),
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/caching/caching_tests.cpp:87]: cpu? or common functions?

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/caching/caching_tests.cpp [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        auto nms = std::make_shared<op::internal::NmsStaticShapeIE<ov::op::v8::MatrixNms>>(boxes, scores, ov::op::v8::MatrixNms::Attributes());
+        auto res = std::make_shared<ngraph::opset6::Result>(nms);
+        auto func = std::make_shared<Function>(NodeVector{nms}, ParameterVector{boxes, scores});
+        return func;
+    }
+
+    static std::shared_ptr<ngraph::Function> simple_function_multiclass_nms_internal(ngraph::element::Type, size_t) {
+        auto boxes = std::make_shared<opset1::Parameter>(element::f32, Shape{1, 1000, 4});
+        auto scores = std::make_shared<opset1::Parameter>(element::f32, Shape{1, 1, 1000});
+        auto nms = std::make_shared<op::internal::NmsStaticShapeIE<ov::op::v8::MulticlassNms>>(boxes, scores, ov::op::v8::MulticlassNms::Attributes());
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/caching/caching_tests.cpp:62]: I suppose we need to use public opset in conformance

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/skip_configs/skip_config_cpu.lst [hunk 6]
```diff
@@ -0,0 +1 @@
+.*Auto.*InferRequestPreprocessTest.*SetScalePreProcessSetBlob.*
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/skip_configs/skip_config_cpu.lst:1]: do we have a bug? does it work on MULTI, but fails on AUTO?

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp (hunk 2 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp:71]: typo, thanks

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp (hunk 3 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp:34]: image

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/skip_configs/skip_config_cpu.lst (hunk 6 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/skip_configs/skip_config_cpu.lst:1]: Will be removed

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp (hunk 3 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp:34]: image

File: inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/CMakeLists.txt (hunk 1 above)
Comment (iefode) [reply inference-engine/tests/functional/plugin/conformance/test_runner/api_conformance_runner/CMakeLists.txt:11]: will be fixed, thanks

**Comment by iefode** [comment]
@ilya-lavrenov All comments were applied. Please review the PR again :)

**Review by andrei-kochin — APPROVED** [review]

**Pull request merged**